packages/react-components/react-combobox/src/components/Combobox/useCombobox.tsx

```diff
@@ -170,28 +159,31 @@ export const useCombobox_unstable = (props: ComboboxProps, ref: React.Ref<HTMLIn
   const resolvedPropsOnKeyDown = triggerSlot.onKeyDown;
   triggerSlot.onChange = mergeCallbacks(triggerSlot.onChange, onTriggerChange);
   triggerSlot.onBlur = mergeCallbacks(triggerSlot.onBlur, onTriggerBlur); // only resolve listbox slot if needed
-  listboxSlot =
-    open || hasFocus
-      ? slot.optional(props.listbox, {
-          renderByDefault: true,
-          defaultProps: { children: props.children, style: popupDimensions },
-          elementType: Listbox,
-        })
-      : undefined;
-  [triggerSlot, listboxSlot] = useComboboxPopup(props, triggerSlot, listboxSlot);
+  listboxSlot = slot.optional(props.listbox, {
+    renderByDefault: open || hasFocus,
+    defaultProps: { children: props.children },
+    elementType: Listbox,
+  });
+  const rootSlot = slot.always(props.root, {
+    defaultProps: {
+      'aria-owns': !inlinePopup ? listboxSlot?.id : undefined,
+      ...rootNativeProps,
+    },
+    elementType: 'div',
+  });
+  rootSlot.ref = useMergedRefs(rootSlot.ref, comboboxTargetRef);
```

Comment:
There's actually no requirement that the positioned target has to be the 'trigger'. Moving this to the root seems more appropriate since the previous logic always use the root as the desired dimension